WordPress/Classes/ViewRelated/Plans/PlanDetailViewController.swift

```diff
@@ -150,42 +148,47 @@ class PlanDetailViewController: UIViewController {
 
         return controller
     }
-    
+
     override func prefersStatusBarHidden() -> Bool {
         return true
     }
-    
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        
+
         configureAppearance()
         configureTableView()
         populateHeader()
         updateNoResults()
     }
     
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        
+        let size = headerView.systemLayoutSizeFittingSize(UILayoutFittingCompressedSize)
+        headerView.frame.size.height = size.height
+        tableView.tableHeaderView = headerView
+        view.layoutIfNeeded()
```

Comment:
Thanks for the comments here – I've moved this so it's only executed whenever the header is repopulated, and I've added a comment to explain why it's necessary.